anndata/_io/utils.py

```diff
@@ -151,31 +153,26 @@ class AnnDataReadError(OSError):
     pass
 
 
-def _get_parent(elem):
-    try:
-        import zarr
-    except ImportError:
-        zarr = None
-    if zarr and isinstance(elem, (zarr.Group, zarr.Array)):
-        parent = elem.store  # Not sure how to always get a name out of this
-    elif isinstance(elem, BaseCompressedSparseDataset):
-        parent = elem.group.file.name
-    else:
-        parent = elem.file.name
-    return parent
+def _get_path(elem: Elem) -> str:
+    """Return an absolute path of an element (always starts with “/”)."""
+    if isinstance(elem, BaseCompressedSparseDataset):
+        elem = elem.group
+    path = elem.name or "??"  # can be None
```

Comment:
This function is called for display purposes, so I’m handling the `None` here instead of passing it on and handling it on the call site.

I’m also not entirely sure that this case can happen at all, but I added this defensive path in case it does